corehq/apps/fixtures/models.py

```diff
@@ -289,6 +289,38 @@ def _serialize(val):
 
         return xData
 
+    def _get_reporting_groups(self, group_ids):
+        groups = []
+
+        reporting_group_ids = set([
+            gid for gid in group_ids
+            if gid.startswith(LOCATION_REPORTING_PREFIX)
+        ])
+        reporting_location_ids = [
+            group_id[group_id.index('-') + 1:]
+            for group_id in reporting_group_ids
+        ]
+        reporting_locations = SQLLocation.objects.filter(
+            location_id__in=reporting_location_ids
+        )
+
+        for reporting_location in reporting_locations:
+            groups.append(reporting_location.reporting_group_object())
+
+        return groups
+
+    def _get_case_sharing_groups(self, group_ids):
+        groups = []
+
+        case_sharing_locations = set(SQLLocation.objects.filter(
+            location_id__in=group_ids
+        ).values_list('location_id', flat=True))
```

Comment:
I'm sure packing a list into a set (then pulling it back out) is slower than relying on SQL, so fixed that.